jarviscli/plugins/music_recognition.py

```diff
@@ -0,0 +1,210 @@
+from io import BytesIO
+import tempfile
+import urllib.error
+import urllib.request
+from plugin import plugin, require, alias
+from colorama import Fore
+from shazamio import Shazam
+import pydub
+import asyncio
+import climage
+
+requirements = ['ffmpeg']
+try:
+    import pyaudio
+    import speech_recognition as sr
+except ImportError:
+    requirements.append(
```

Comment:
Pretty clever, I like this idea. Yeah, this is better than anything I thought of; I'll use this pattern for other plugins, too ;).